libr/parse/p/parse_6502_pseudo.c

```diff
@@ -88,13 +84,14 @@ static int replace(int argc, const char *argv[], char *newstr, ADDR_TYPE type) {
 						strcpy(newstr + k, w);
 						k += strlen(w) - 1;
 					}
-				} else
+				} else {
 					newstr[k] = ops[i].str[j];
+				}
 			}
 			newstr[k] = '\0';
 			if (argc == 4 && argv[2][0] == '[') {
-				strcat(newstr + k, "+");
-				strcat(newstr + k + 3, argv[3]);
+				strcat (newstr + k, "+");
+				strcat (newstr + k + 3, argv[2]);
```

Comment:
check this @ricardoquesada please :). This function is called in `parse`. The maximum index for wa is 2 and in `replace` was using 3 causing a oob read. Check if this break something though running the regression test t.esil/6502 i get green
```
const char *wa[] = {w0, w1, w2};                                                                                                                                                      
int nw = 0;                                                                                                                                                                           
for (i = 0; i < 3; i++) {                                                                                                                                                             
             if (wa[i][0]) nw++;                                                                                                                                                           
}                                                                                                                                                                                     
replace (nw, wa, str, atype);
```